src/base/UnorderedNonce.sol

```diff
@@ -12,21 +12,42 @@ contract UnorderedNonce {
     /// @param owner address, the owner/signer of the nonce
     /// @param nonce uint256, the nonce to consume
     function _useUnorderedNonce(address owner, uint256 nonce) internal {
-        (uint256 wordPos, uint256 bitPos) = bitmapPositions(nonce);
-        uint256 bit = 1 << bitPos;
-        uint256 flipped = nonces[owner][wordPos] ^= bit;
-
-        if (flipped & bit == 0) revert NonceAlreadyUsed();
+        // consume the bit by flipping it in storage
+        // reverts if the bit was already spent
+        _flipBit(nonces[owner], nonce);
     }
 
-    /// @notice Returns the index of the bitmap and the bit position within the bitmap. Used for unordered nonces
-    /// @param nonce The nonce to get the associated word and bit positions
-    /// @return wordPos The word position or index into the nonceBitmap
-    /// @return bitPos The bit position
-    /// @dev The first 248 bits of the nonce value is the index of the desired bitmap
-    /// @dev The last 8 bits of the nonce value is the position of the bit in the bitmap
-    function bitmapPositions(uint256 nonce) private pure returns (uint256 wordPos, uint256 bitPos) {
-        wordPos = uint248(nonce >> 8);
-        bitPos = uint8(nonce);
+    function _flipBit(mapping(uint256 => uint256) storage bitmap, uint256 nonce) private {
```

Comment:
imo this does more than flip the bit so we should maybe split out this code.

1 - use the nonce to extract the wordPos/the bitmap to operate on 
2- flip bit

flipBit(nonce) makes it seem like the nonce thats being passed in is the bit thats being flipped which is not really how the data is used